nailgun/entities.py

```diff
@@ -544,6 +544,58 @@ def create_payload(self):
         }
 
 
+class DiscoveredHosts(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntityUpdateMixin):
+    """A representation of a Foreman Discovered Host entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(required=True),
+            'ip': entity_fields.IPAddressField(required=True),
+            'mac': entity_fields.MACAddressField(required=True),
+        }
+        self._meta = {
+            'api_path': '/api/v2/discovered_hosts',
+            'server_modes': ('sat'),
+        }
+        super(DiscoveredHosts, self).__init__(server_config, **kwargs)
+
+    def create_payload(self):
+        """Wrap submitted data within an extra dict.
+
+        For more information, see `Bugzilla #1151220
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1151220>`_.
+
+        """
+        return {
+            u'discovered_host': super(
+                DiscoveredHosts,
+                self
+            ).create_payload()
+        }
+
+    def upload_facts(self, params):
```

Comment:
nitpick: use `payload` word instead `params`. We had huge re-factor process to move to that word for *all* helper methods